app/code/Magento/ConfigurableProduct/view/frontend/web/js/configurable.js

```diff
@@ -387,14 +394,27 @@ define([
                         }
                     } else {
                         allowedProducts = options[i].products.slice(0);
+                        if (typeof allowedProducts[0] !== "undefined"
+                            && typeof optionPrices[allowedProducts[0]] !== "undefined") {
+
+                            optionFinalPrice = parseFloat(optionPrices[allowedProducts[0]].finalPrice.amount);
+                            optionPriceDiff = optionFinalPrice - basePrice;
+
+                            if (optionPriceDiff !== 0) {
+                                options[i].label = options[i].label + ' ' + priceUtils.formatPrice(
+                                    optionPriceDiff,
+                                    this.options.priceFormat,
+                                    1);
+                            }
+                        }
                     }
 
                     if (allowedProducts.length > 0) {
                         options[i].allowedProducts = allowedProducts;
                         element.options[index] = new Option(this._getOptionLabel(options[i]), options[i].id);
 
                         if (typeof options[i].price !== 'undefined') {
```

Comment:
Looks like this functionality is never used. Also it has a bug with attribute name (uses price**s** instead price)